src/astra_rl/core/environment.py

```diff
@@ -0,0 +1,36 @@
+"""
+rollout.py
+Roll out a problem
+"""
+
+from abc import abstractmethod, ABC
+from dataclasses import dataclass
+from typing import Sequence, Generic, Self, Optional
+
+from astra_rl.core.common import StateT, ActionT
+from astra_rl.core.problem import Problem
+
+
+@dataclass
+class Node(Generic[StateT, ActionT]):
```

Comment:
Would be useful to do a scrub of the PR for everything that's a definition to add docstrings for the class and method definitions to capture the context and information from the implementation and reasoning while it's all still fresh.